Kernel/Memory/MemoryManager.cpp

```diff
@@ -243,7 +243,9 @@ UNMAP_AFTER_INIT void MemoryManager::parse_memory_map()
     // Register used memory regions that we know of.
     m_global_data.with([&](auto& global_data) {
         global_data.used_memory_ranges.ensure_capacity(4);
+#if ARCH(I386) || ARCH(X86_64)
         global_data.used_memory_ranges.append(UsedMemoryRange { UsedMemoryRangeType::LowMemory, PhysicalAddress(0x00000000), PhysicalAddress(1 * MiB) });
```

Comment:
Is it not necessary to map the zero page, to make null pointer derefs fault?

Reply:
The `used_memory_ranges` are physical memory ranges that are already occupied in physical memory. For example for `x86` the 0-1Mib physical range is used for BIOS related things IIRC. However the kernel is actually running in the virtual memory space, where the zero page currently is unmapped, thus causing a fault when it is accessed.